### PR TITLE
Update CryptoPP & Argon2

### PR DESCRIPTION
**Pull request by SergiySW**
Also:
* move external crypto libraries to crypto folder
* add xxHash as a submodule

### CODE REVIEW

**Comment by SergiySW** [comment]
Travis build fails because of clang-format check of old good Blake2

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,13 +54,15 @@ else ()
 
 	IF (CMAKE_SYSTEM_PROCESSOR MATCHES "^(i.86|x86(_64)?)$")
 		if (NANO_SIMD_OPTIMIZATIONS OR RAIBLOCKS_SIMD_OPTIMIZATIONS)
-			add_compile_options(-msse4)
+			add_compile_options(-msse4 -maes)
```
Comment (cryptocode) [inline CMakeLists.txt]: Should there be a NANO_ENABLE_AES flag? Seems like a cpu can have SSE4 support but not the AES instruction set. The -maes use later on inside ENABLE_AVX2 is probably fine.

Comment (SergiySW) [reply CMakeLists.txt]: But we didn't document that SIMD optimizations = SSE4.2 only

Comment (cryptocode) [reply CMakeLists.txt]: Should we add docs that AES support is required for SIMD optimizations then? For instance, there are corei7's with sse4, but not avx/aes.

Comment (SergiySW) [reply CMakeLists.txt]: Enabled by default for SIMD now, but can be disabled if required

Comment (cryptocode) [reply CMakeLists.txt]: Perfect

**Comment by rkeene** [comment]
Travis CI picked up a bunch of formatting errors issues, probably related to the new code location not being blacklisted.  Can you get it to pass the automated tests ?

**Comment by rkeene** [comment]
```
+gtimeout 120 ./core_test
./ci/test.sh: line 53: 89254 Segmentation fault: 11  ${TIMEOUT_CMD} ${TIMEOUT_TIME_ARG} ${TIMEOUT_SEC-${TIMEOUT_DEFAULT}} ./core_test
+core_test_res=139
+'[' 139 = 0 ']'
+for try in '"${tries[@]}"'
+'[' 5 '!=' _initial_ ']'
+echo 'core_test failed: 139, retrying (try=5)'
core_test failed: 139, retrying (try=5)
+sleep 68
+gtimeout 120 ./core_test
./ci/test.sh: line 53: 89257 Segmentation fault: 11  ${TIMEOUT_CMD} ${TIMEOUT_TIME_ARG} ${TIMEOUT_SEC-${TIMEOUT_DEFAULT}} ./core_test
+core_test_res=139
+'[' 139 = 0 ']'
+for try in '"${tries[@]}"'
+'[' 6 '!=' _initial_ ']'
+echo 'core_test failed: 139, retrying (try=6)'
core_test failed: 139, retrying (try=6)
+sleep 86
+gtimeout 120 ./core_test
./ci/test.sh: line 53: 89260 Segmentation fault: 11  ${TIMEOUT_CMD} ${TIMEOUT_TIME_ARG} ${TIMEOUT_SEC-${TIMEOUT_DEFAULT}} ./core_test
```

**Pull request merged**